six/modules/c++/six/include/six/NITFWriteControl.h

```diff
@@ -78,7 +78,7 @@ class NITFWriteControl : public WriteControl
     // Get the container that we were initialized from (const)
     const Container* getContainer() const
     {
-        return mContainer;
+        return mContainer.get();
```

Comment:
Shouldn't this signature change to match the ReadControl and return a shared pointer?  Also probably move it to the base class for consistency.